upf/upf_api.c

```diff
@@ -1061,8 +1061,39 @@ vl_api_upf_tdf_ul_table_add_t_handler (vl_api_upf_tdf_ul_table_add_t * mp)
   REPLY_MACRO (VL_API_UPF_TDF_UL_TABLE_ADD_REPLY);
 }
 
-#include <upf/upf.api.c>
+/* API message handler */
+static void
+vl_api_upf_nat_pool_add_t_handler (vl_api_upf_nat_pool_add_t * mp)
+{
+  int rv = 0;
+  upf_main_t *sm = &upf_main;
+  vl_api_upf_nat_pool_add_reply_t *rmp = NULL;
+
+  u8 *nwi_s = mp->names;
+  u8 *nwi_s_vec = 0;
+  u8 *name = mp->names;
+  u8 *nwi_name;
+  ip4_address_t start, end;
+  u16 min_port = UPF_NAT_MIN_PORT;
+  u16 max_port = UPF_NAT_MAX_PORT;
+  u32 port_block_size;
+  u8 is_add = mp->is_add;
 
+  // calculate start of second cstring
+  while (*name++);
```

Comment:
This will fail in an unpredictable way if `name` doesn't happen to contain a zero byte. Please check for this situation